src/nevermined/api/RegistryBaseApi.ts

```diff
@@ -586,6 +586,17 @@ export abstract class RegistryBaseApi extends Instantiable {
         throw new AssetError(`Error creating ${serviceType} agreement`)
       }
 
+      // Checking the agreementId was created on-chain with the correct DID associated to it
+      const agreementData = await this.nevermined.keeper.agreementStoreManager.getAgreement(
```

Comment:
`agreementStoreManager.getAgreement` is using events. We should call https://github.com/nevermined-io/contracts/blob/main/contracts/templates/BaseEscrowTemplate.sol#LL173 to check if an agreement exists without using events